### PR TITLE
added stringification support

### PR DESCRIPTION
**Pull request by jonnybi**
Added "force stringification" support as described in [serilog documentation](https://github.com/serilog/serilog/wiki/Structured-Data).

### CODE REVIEW

File: src/Serilog.Extensions.Logging/Extensions/Logging/SerilogLogger.cs [hunk 1]
```diff
@@ -78,6 +78,11 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                         if (logger.BindProperty(property.Key.Substring(1), property.Value, true, out var destructured))
                             properties.Add(destructured);
                     }
+                    else if (property.Key.StartsWith("$"))
+                    {
+                        if (logger.BindProperty(property.Key.Substring(1), property.Value?.ToString() ?? "null", true, out var stringified))
```
Comment (nblumhardt) [inline src/Serilog.Extensions.Logging/Extensions/Logging/SerilogLogger.cs]: Substituting in the text `null` will be inconsistent with Serilog here, I think. We should just pass null if `property.Value` is null.

**Comment by nblumhardt** [comment]
@jonnybi hi! I just spotted one more place where we special-case `@`, which I think should also accept `$` for the sake of consistency:

https://github.com/serilog/serilog-extensions-logging/blob/dev/src/Serilog.Extensions.Logging/Extensions/Logging/SerilogLoggerScope.cs#L75

**Comment by jonnybi** [comment]
@nblumhardt  thx for spotting this. i will check it

File: src/Serilog.Extensions.Logging/Extensions/Logging/SerilogLoggerScope.cs [hunk 2]
```diff
@@ -78,6 +78,11 @@ public void EnrichAndCreateScopeItem(LogEvent logEvent, ILogEventPropertyFactory
                         destructureObject = true;
                     }
 
+                    if (key.StartsWith("$"))
+                    {
+                        key = key.Substring(1);
```
Comment (nblumhardt) [inline src/Serilog.Extensions.Logging/Extensions/Logging/SerilogLoggerScope.cs]: The value would also have to be `ToString()`ed in order for this to take effect.

Comment (jonnybi) [reply src/Serilog.Extensions.Logging/Extensions/Logging/SerilogLoggerScope.cs]: sry, i was a little too fast. Fixed it again and added a unit test.

**Comment by nblumhardt** [comment]
Thanks! 👍

**Pull request merged**